src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs

```diff
@@ -473,6 +473,31 @@ private async Task WriteExecutableChange(ExecutableViewModel executableViewModel
                 resourceViewModel.Endpoints.Add(endpointString);
             }
         }
+
+        var resourceApplicationModel = applicationModel.Resources.FirstOrDefault(r => r.Name == resourceViewModel.Name);
+        if (resourceApplicationModel?.TryGetAllocatedEndPoints(out var allocatedEndPoints) ?? false)
+        {
+            if (allocatedEndPoints.Count() == 1)
+            {
+                var service = services.FirstOrDefault(s => s.Metadata.Name == resourceViewModel.Name);
+                if (service != null)
+                {
+                    resourceViewModel.Services.Add(new ResourceService(service.Metadata.Name, service.AllocatedAddress, service.AllocatedPort));
+                }
+            }
+            else
+            {
+                foreach (var allocatedEndPoint in allocatedEndPoints)
+                {
+                    var serviceName = resourceApplicationModel.Name + "_" + allocatedEndPoint.Name;
```

Comment:
I noticed that if there is one service for a resource then its name matches the resource. e.g. `frontend` service for the `frontend` resource. But if there are two then the name is added to the service, e.g. `postgres_tcp` and `postgres_management` for the `postgres` resource.

(although it occurs to me that maybe the service name is always appended, and in eShopLite a name is only added when there are multiple services)

Hosting experts: I'd like your guidance to improve this code.